### PR TITLE
community: Fixing a performance issue with AzureSearch to perform batch embedding

### PR DESCRIPTION
**Pull request by msetbar**
- **Description:** Azure Cognitive Search vector DB store performs slow embedding as it does not utilize the batch embedding functionality. This PR provide a fix to improve the performance of Azure Search class when adding documents to the vector search, 
  - **Issue:** #11313 ,
  - **Dependencies:** any dependencies required for this change,
  - **Twitter handle:** we announce bigger features on Twitter. If your PR gets announced, and you'd like a mention, we'll gladly shout you out!

Please make sure your PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` from the root of the package you've modified to check this locally.

See contribution guidelines for more information on how to write/run tests, lint, etc: https://python.langchain.com/docs/contributing/

If you're adding a new integration, please include:
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

If no one reviews your PR within a few days, please @-mention one of @baskaryan, @eyurtsev, @hwchase17.
 -->

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -329,6 +335,20 @@ def add_texts(
         """Add texts data to an existing index."""
         keys = kwargs.get("keys")
         ids = []
+
+        # batching support if embedding function is an Embeddings object
+        if isinstance(self.embedding_function, Embeddings):
+            try:
+                embeddings = self.embedding_func.embed_documents(texts)
```
Comment (butlfrazp) [inline libs/community/langchain_community/vectorstores/azuresearch.py:342]: should this be `embedding_function`? Same below

**Review by butlfrazp — COMMENTED** [review]
I think in the `from_texts` classmethod we can update the embeddings parameter.
I think there are some other instances where the embedding_function is used but not checked for (ex. line 532)

File: libs/community/langchain_community/vectorstores/azuresearch.py (hunk 1 above)
Comment (msetbar) [reply libs/community/langchain_community/vectorstores/azuresearch.py]: fixed.

**Comment by msetbar** [comment]
@baskaryan, @eyurtsev, @hwchase17 could you please advise on the next step on this PR?

**Review by butlfrazp — APPROVED** [review]

**Comment by msetbar** [comment]
@hwchase17 @eyurtsev @baskaryan could you please help with this?

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**